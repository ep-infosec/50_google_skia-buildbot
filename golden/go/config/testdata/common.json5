{
  // This is a comment to prove this is a JSON5 format.
  common_str: "somestring",
  common_int: 1234,
  common_bool: false,

  will_be_overridden: "56",

  unknown_field_will_be_ignored: "super ignored",
}
