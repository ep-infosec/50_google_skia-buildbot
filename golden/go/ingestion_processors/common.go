package ingestion_processors

import (
	"context"
	"encoding/json"
	"io"

	"go.opencensus.io/trace"

	"go.skia.org/infra/go/skerr"
	"go.skia.org/infra/go/sklog"
	"go.skia.org/infra/go/util"
	"go.skia.org/infra/go/vcsinfo"
	"go.skia.org/infra/golden/go/jsonio"
)

// parseGoldResultsFromReader parses the JSON stream out of the io.ReadCloser
// into a jsonio.GoldResults instance and closes the reader.
func parseGoldResultsFromReader(r io.ReadCloser) (*jsonio.GoldResults, error) {
	defer util.Close(r)

	// Parse the bytes from the reader as JSON and validate it.
	gr := &jsonio.GoldResults{}
	if err := json.NewDecoder(r).Decode(gr); err != nil {
		return nil, skerr.Wrapf(err, "could not parse JSON")
	}
	if err := gr.UpdateLegacyFields(); err != nil {
		return nil, skerr.Wrap(err)
	}
	if err := gr.Validate(); err != nil {
		return nil, skerr.Wrap(err)
	}
	return gr, nil
}

// processGoldResults opens the given JSON input file and processes it, converting
// it into a jsonio.GoldResults object and returning it. It will close the file when done.
func processGoldResults(ctx context.Context, r io.ReadCloser) (*jsonio.GoldResults, error) {
	ctx, span := trace.StartSpan(ctx, "ingestion_processGoldResults")
	defer span.End()
	defer util.Close(r)
	gr, err := parseGoldResultsFromReader(r)
	if err != nil {
		return nil, skerr.Wrap(err)
	}
	return gr, nil
}

// getCanonicalCommitHash returns the commit hash in the primary repository. If the given
// target hash is not in the primary repository it will try and find it in the secondary
// repository which has the primary as a dependency.
func getCanonicalCommitHash(ctx context.Context, vcs vcsinfo.VCS, targetHash string) (string, error) {
	if isCommit(ctx, vcs, targetHash) {
		return targetHash, nil
	}
	// TODO(kjlubick) We need a way to handle secondary repos (probably not something that
	//   clutters the VCS interface). skbug.com/9628
	sklog.Warningf("Unable to find commit %s in primary repo and no secondary configured", targetHash)

	c := vcs.LastNIndex(3)
	if len(c) == 3 {
		sklog.Debugf("Last three commits were %s on %s, %s on %s, and %s on %s",
			c[0].Hash, c[0].Timestamp, c[1].Hash, c[1].Timestamp, c[2].Hash, c[2].Timestamp)
	} else {
		sklog.Debugf("Last commits: %v", c)
	}
	return "", skerr.Fmt("Unknown commit: %s", targetHash)
}

// isCommit returns true if the given commit is in vcs.
func isCommit(ctx context.Context, vcs vcsinfo.VCS, commitHash string) bool {
	ret, err := vcs.Details(ctx, commitHash, false)
	if err != nil {
		// Let's try updating the VCS - perhaps this is a new commit?
		if err := vcs.Update(ctx, true, false); err != nil {
			sklog.Errorf("Could not update VCS: %s", err)
			return false
		}
		ret, err = vcs.Details(ctx, commitHash, false)
	}
	return err == nil && ret != nil
}
