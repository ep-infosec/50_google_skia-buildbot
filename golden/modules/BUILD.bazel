load("//infra-sk:index.bzl", "karma_test", "sass_library", "ts_library")

karma_test(
    name = "common_test",
    src = "common_test.ts",
    deps = [
        ":common_ts_lib",
        "//golden/modules/search-controls-sk",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "common_ts_lib",
    srcs = ["common.ts"],
    visibility = ["//visibility:public"],
    deps = [
        ":rpc_types_ts_lib",
        "//golden/modules/search-controls-sk",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "demo_util_ts_lib",
    srcs = ["demo_util.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "rpc_types_ts_lib",
    srcs = ["rpc_types.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "settings_ts_lib",
    srcs = ["settings.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "shared_demo_data_ts_lib",
    srcs = ["shared_demo_data.ts"],
    visibility = ["//visibility:public"],
    deps = ["@npm//common-sk"],
)

sass_library(
    name = "colors_sass_lib",
    srcs = ["colors.scss"],
    visibility = ["//visibility:public"],
)
