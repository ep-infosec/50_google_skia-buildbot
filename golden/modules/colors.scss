/* Common Gold Colors via CSS Variables.

 Can be imported via:

    @import '../colors';
*/

html {
  --gold: rgb(217, 95, 2);
  --bright-orange: rgb(255, 127, 0);

  --light-error-red: rgb(255, 230, 230);
  --transparent-white: rgba(255, 255, 255, 0);

  --triaged-positive: #1B9E77;
  --triaged-negative: #E7298A;
  --untriaged: #A6761D;
}
