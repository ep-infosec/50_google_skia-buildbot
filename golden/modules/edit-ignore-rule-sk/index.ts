import './edit-ignore-rule-sk';
