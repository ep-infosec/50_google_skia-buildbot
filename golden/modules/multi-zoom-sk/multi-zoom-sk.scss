@import 'node_modules/elements-sk/colors.scss';

multi-zoom-sk {
  font-family: sans-serif;

  > .container {
    min-width: 700px;
    width: inherit;
    min-height: 500px;
    display: flex;
    flex-direction: row;
  }

  .preview_and_zoomed {
    display: flex;
    flex-direction: column;
    padding: 0.5em;
  }

  .previews_and_toggles {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  .preview {
    position: relative;
    margin: 0;
    img.thumbnail {
      border: solid 2px lightgray;
      margin: 0;
      display: block;
      width: 128px;
      height: 128px;
      object-fit: scale-down;
      object-position: top left;
    }
    // position this exactly over the thumbnail.
    canvas.crosshair {
      position: absolute;
      top: 0;
      left: 0;
      margin: 2px;
      width: 128px;
      height: 128px;
    }

    figcaption {
      checkbox-sk {
        margin: 3px 0;
      }
      .displayed {
        font-weight: bold;
      }
      .for_spacing {
        // See https://stackoverflow.com/a/20249560 Basically, we have a copy of
        // the checkbox with label that is hidden and bolded so the width doesn't
        // jump around when it alternates between displayed and not displayed.
        height: 0;
        overflow: hidden;
        margin: 0;
      }
    }
  }

  .size_warning {
    font-weight: bold;
    color: var(--red);
    margin-bottom: 8px;
  }

  .zoomed_view {
    canvas {
      border: 1px solid black;
      width: 500px;
      height: 500px;
    }
  }

  .stats_and_nav {
    td.label {
      font-weight: bold;
    }
    td.value {
      font-family: monospace;
    }
    th {
      text-align: left;
    }

    table {
      margin-bottom: 10px;
    }
  }

  canvas.scratch {
    display: none;
  }

}
