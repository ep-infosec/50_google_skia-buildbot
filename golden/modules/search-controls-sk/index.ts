import './search-controls-sk';

export { SearchCriteriaToHintableObject, SearchCriteriaFromHintableObject } from './search-controls-sk';
