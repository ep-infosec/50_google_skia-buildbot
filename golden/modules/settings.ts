/**
 * This file lists helpers for returning global constants in Gold. These should not change once
 * an instance has booted up, so they need not be returned via JSON requests.
 *
 * Settings are expected to be in window.GoldSettings and the functions here are nice helpers
 * for that, so as to demystify "where do these values come from?"
 */

export interface GoldSettings {
  title?: string;
  defaultCorpus?: string;
  baseRepoURL?: string;
}

function getSettings(): GoldSettings | undefined {
  return (window as any).GoldSettings as GoldSettings | undefined;
}

export function title(): string {
  return getSettings()?.title || '';
}

export function defaultCorpus(): string {
  return getSettings()?.defaultCorpus || '';
}

export function baseRepoURL(): string {
  return getSettings()?.baseRepoURL || '';
}

export function testOnlySetSettings(newSettings: GoldSettings) {
  (window as any).GoldSettings = newSettings;
}
