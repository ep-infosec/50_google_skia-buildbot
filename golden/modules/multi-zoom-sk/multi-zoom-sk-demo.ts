import './index';

import { $$ } from 'common-sk/modules/dom';
import { digestDiffImagePath, digestImagePath } from '../common';
import { diff16x16, left16x16, right16x16 } from './test_data';
import { isPuppeteerTest } from '../demo_util';
import { MultiZoomSk } from './multi-zoom-sk';

const isPuppeteer = isPuppeteerTest();

let ele = new MultiZoomSk();
ele.cyclingView = !isPuppeteer;
ele.details = {
  leftImageSrc: digestImagePath('99c58c7002073346ff55f446d47d6311'),
  diffImageSrc: digestDiffImagePath(
    '99c58c7002073346ff55f446d47d6311', '6246b773851984c726cb2e1cb13510c2',
  ),
  rightImageSrc: digestImagePath('6246b773851984c726cb2e1cb13510c2'),
  leftLabel: '99c58c700207...',
  rightLabel: 'Closest Positive',
};
// These coordinates show an interesting difference.
ele.x = 77;
ele.y = 199;
$$<HTMLDivElement>('#normal')!.appendChild(ele);

ele = new MultiZoomSk();
ele.cyclingView = !isPuppeteer;
ele.details = {
  leftImageSrc: digestImagePath('ec3b8f27397d99581e06eaa46d6d5837'),
  diffImageSrc:
      digestDiffImagePath('ec3b8f27397d99581e06eaa46d6d5837', '6246b773851984c726cb2e1cb13510c2'),
  rightImageSrc: digestImagePath('6246b773851984c726cb2e1cb13510c2'),
  leftLabel: 'ec3b8f27397...',
  rightLabel: '6246b773851...',
};
$$<HTMLDivElement>('#mismatch')!.appendChild(ele);

ele = new MultiZoomSk();
ele.cyclingView = !isPuppeteer;
ele.details = {
  leftImageSrc: left16x16,
  diffImageSrc: diff16x16,
  rightImageSrc: right16x16,
  leftLabel: 'left16x16',
  rightLabel: 'right16x16',
};
$$<HTMLDivElement>('#base64')!.appendChild(ele);

ele = new MultiZoomSk();
ele.cyclingView = !isPuppeteer;
ele.details = {
  leftImageSrc: digestImagePath('99c58c7002073346ff55f446d47d6311'),
  diffImageSrc:
      digestDiffImagePath('99c58c7002073346ff55f446d47d6311', '6246b773851984c726cb2e1cb13510c2'),
  rightImageSrc: digestImagePath('6246b773851984c726cb2e1cb13510c2'),
  leftLabel: '99c58c700207...',
  rightLabel: 'Closest Positive',
};
ele.zoomLevel = 32;
ele.showGrid = true;
// These coordinates show an interesting difference.
ele.x = 77;
ele.y = 199;
$$<HTMLDivElement>('#zoomed_grid')!.appendChild(ele);

const nthPixel = new MultiZoomSk();
nthPixel.addEventListener('sources-loaded', () => {
  nthPixel.moveToNextLargestDiff(false);
});
nthPixel.cyclingView = !isPuppeteer;
nthPixel.details = {
  leftImageSrc: left16x16,
  diffImageSrc: diff16x16,
  rightImageSrc: right16x16,
  leftLabel: 'left16x16',
  rightLabel: 'right16x16',
};

$$<HTMLDivElement>('#base64_nthpixel')!.appendChild(nthPixel);

// This element demonstrates how to use multi-zoom-sk in a dialog. It is not meant for use on
// puppeteer.
$$<MultiZoomSk>('#in_dialog')!.details = {
  leftImageSrc: digestImagePath('99c58c7002073346ff55f446d47d6311'),
  diffImageSrc:
      digestDiffImagePath('99c58c7002073346ff55f446d47d6311', '6246b773851984c726cb2e1cb13510c2'),
  rightImageSrc: digestImagePath('6246b773851984c726cb2e1cb13510c2'),
  leftLabel: '99c58c700207...',
  rightLabel: 'Closest Positive',
};

$$<HTMLButtonElement>('#dialog_btn')!.addEventListener('click', () => {
  $$<HTMLDialogElement>('#the_dialog')!.showModal();
});

$$<HTMLButtonElement>('#close_btn')!.addEventListener('click', () => {
  $$<HTMLDialogElement>('#the_dialog')!.close();
});
