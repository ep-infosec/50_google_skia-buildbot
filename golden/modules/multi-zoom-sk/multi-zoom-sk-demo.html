<!DOCTYPE html>
<html>
  <head>
    <title>multi-zoom-sk</title>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <style>
      .demo_container {
        width: 900px;
      }
    </style>
  </head>
  <body class="body-sk">
    <button id="dialog_btn">Push for Dialog</button>

    <h2>Image sizes match</h2>
    <div id="normal" class="demo_container"></div>

    <h2>Image sizes mismatch</h2>
    <div id="mismatch" class="demo_container"></div>

    <h2>Small base64 image</h2>
    <div id="base64" class="demo_container"></div>

    <h2>Zoomed in with the grid</h2>
    <div id="zoomed_grid" class="demo_container"></div>

    <h2>Jump to closest pixel diff</h2>
    <div id="base64_nthpixel" class="demo_container"></div>

    <dialog id="the_dialog">
      <multi-zoom-sk id="in_dialog"></multi-zoom-sk>
      <button id="close_btn">Close</button>
    </dialog>
  </body>
</html>
